internal/config/config.go

```diff
@@ -86,6 +87,7 @@ type DatabaseConfig struct {
 
 type StorageConfig struct {
 	DataDir string `env:"DIR"` // Using DIR to be backwards compatible with the old config
+	MaxParDl int `env:"MAX_PAR_DL,default=-1"`
```

Comment:
Why `-1` instead of just `0`, the default? In order to block downloads?

Reply:
At this point I no longer remember why I thought `-1` was more appropriate. I guess I thought of `0` as a valid value to pass, but that doesn't really make sense if that's already commonly thought to mean "I trust the program to make a good judgement".
I'll change it to `0`.

Reply:
I think I remember why I used `-1` and it's effectively not a very good reason. I thought "`0` will probably be used to mean 'no limit' so I need a different value", but ended up not implementing any kind of 'no limit' mode, mostly because there is no practical way to, given the semaphore is implemented in a way that grows linearly with the number of allowed elements. The alternative would be to put ifs everywhere, which is rather unclean.
Users that want unlimited parallelism can simply pass in a value higher than the number of entries to process or a ridiculously large number I guess.